docs/opentracing.md

```diff
@@ -51,6 +51,9 @@ docker run -d --name jaeger \
   jaegertracing/all-in-one:1
 ```
 
+Additionally to point synapse to a specific IP you can use the
+`JAEGER_AGENT_HOST` environment variable.
+
```

Comment:
Presumably this comes from https://github.com/jaegertracing/jaeger-client-python/blob/5f68f1171c4156fd05675b6782aac1182058e7f4/jaeger_client/config.py#L294-L304.

This feels like it ought to be under the "Enable OpenTracing in Synapse" section below. Putting it here makes it sound like this JAEGER_AGENT_HOST is needed for this simple setup.

Suggest something like:

> By default, Synapse will publish traces to Jaeger on localhost. If Jaeger is hosted elsewhere, point Synapse to the correct host by settings the `JAEGER_AGENT_HOST` environment variable.

Additionally:

- A documentation link would be nice. The best I could see is https://github.com/jaegertracing/jaeger-client-python/pull/218
- Does this work with if you're using https://github.com/erikjohnston/rust-jaeger-python-client? There is code for this, but we don't mark it as a dependency. Do we even support this?
- Is it possible to set something in [`opentracing.jaeger_config`](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#opentracing-1) to do this, for those who are already all-in on using a config file? https://github.com/jaegertracing/jaeger-client-python/blob/5f68f1171c4156fd05675b6782aac1182058e7f4/jaeger_client/config.py#L297 and https://github.com/jaegertracing/jaeger-client-python/blob/5f68f1171c4156fd05675b6782aac1182058e7f4/jaeger_client/config.py#L259C1-L259C1 suggest that 
 ```yaml
 opentracing:
   jaeger_config:
     local_agent:
       reporting_host: 1.2.3.4
   ```
 might work?

Reply:
Sorry yes I should have added links. I originally found it using the source you also mentioned. It does work with synapse but I did not try https://github.com/erikjohnston/rust-jaeger-python-client I think. I am using this together with grafana tempo on my server. 

> Is it possible to set something in [opentracing.jaeger_config](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#opentracing-1) to do this, for those who are already all-in on using a config file?

I can give that a try. It wasn't really obvious if synapse does pass-through/supports their config file here or not.

Reply:
It seems like the rust implementation uses a different variable https://github.com/erikjohnston/rust-jaeger-python-client/blob/master/src/lib.rs#L87-L99

It uses `agent_host_name` from the config only. And it seems to be not a variable that exists in the python version. So it is non-standard I guess.

Not sure if that should be mentioned here.

Reply:
> Is it possible to set something in [opentracing.jaeger_config](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#opentracing-1) to do this, for those who are already all-in on using a config file?

This is the way that we recommend setting jaeger client specific settings.

Reply:
> > Is it possible to set something in [opentracing.jaeger_config](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#opentracing-1) to do this, for those who are already all-in on using a config file?
> 
> This is the way that we recommend setting jaeger client specific settings.

I updated the wording accordingly for that and mention config first, then the env var. Is the text as committed ok?